packages/appsync-modelgen-plugin/src/visitors/appsync-swift-visitor.ts

```diff
@@ -615,9 +615,13 @@ export class AppSyncSwiftVisitor<
     // connected field
     if (connectionInfo) {
       if (connectionInfo.kind === CodeGenConnectionType.HAS_MANY) {
-        return `.hasMany(${name}, is: ${isRequired}, ofType: ${typeName}, associatedWith: ${this.getModelName(
-          connectionInfo.connectedModel,
-        )}.keys.${this.getFieldName(connectionInfo.associatedWith)})`;
+        let connectedModelName = this.getModelName(connectionInfo.connectedModel);
+        const associatedWithAttrStr = this.isCustomPKEnabled()
+          ? `associatedFields: [${connectionInfo.associatedWithFields
+              .map(target => `${connectedModelName}.keys.${this.getFieldName(target)}`)
+              .join(', ')}]`
+          : `associatedWith: ${connectedModelName}.keys.${this.getFieldName(connectionInfo.associatedWith)}`;
+        return `.hasMany(${name}, is: ${isRequired}, ofType: ${typeName}, ${associatedWithAttrStr})`;
```

Comment:
I think this should have impact on `hasMany-belongsTo` bi-direction as well when CPK is true. Could you add a unit test to verify this case?

Reply:
good point, added!